### DISCUSSION TITLE
List of all languages supported by github search

### DISCUSSION THREAD

**Discussion by 0xInfection** [community/community#23724] [New to GitHub]
<p>Hello!</p>
<p>Is there a complete list of languages somewhere which are supported by GitHub search? I would also like to know what are the prefixes for the languages to be used when searching, e.g. - <code>javascript</code> -&gt; <code>js</code> when searching.</p>
<p>Example: <code>language:js someFunc()</code>.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Hello and welcome to the community <a class="mention" href="/0xinfection">@0xInfection</a>.</p>
<p>The <a href="https://github.com/github/linguist">Linguist project</a> is what we use to determine which language a file contains and it has a <a href="https://github.com/github/linguist/blob/b3664e4f242c842e356bd011090f85514d147948/lib/linguist/languages.yml">very exhaustive list of languages it recognizes</a>. Each language can have <a href="https://github.com/github/linguist/blob/b3664e4f242c842e356bd011090f85514d147948/lib/linguist/languages.yml#L2588-L2590">a list of aliases that it recognizes</a> and the <code>languages.yml</code> file is also where to find that information.</p>
<p>Let us know if you have any more questions!</p>

**Reply by DereC4** [reply]
ty

**Comment by 0xInfection** [comment]
<p>Hey <a class="mention" href="/lee-dohm">@lee-dohm</a>  thanks for the great response, very much appreciated. Just a final question, whenever we search GitHub using the web search bar, there is a column at the left side displaying the top 10 languages where the search term has been found. </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/1/d1bf21b42fa21fe36b1437b89a5003a914dff3d3.png" data-download-href="/uploads/short-url/tVvhUWAAfkbUeZI8q3tZrq00um7.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/39941993/181078918-f6e121e2-8397-4b7e-91c3-5b907c6edc0a.png" alt="image" data-base62-sha1="tVvhUWAAfkbUeZI8q3tZrq00um7" width="392" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/1/d1bf21b42fa21fe36b1437b89a5003a914dff3d3_2_392x500.png, https://aws1.discourse-cdn.com/github/original/3X/d/1/d1bf21b42fa21fe36b1437b89a5003a914dff3d3.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/1/d1bf21b42fa21fe36b1437b89a5003a914dff3d3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/1/d1bf21b42fa21fe36b1437b89a5003a914dff3d3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">504×642 29 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
Is there a way I can grab that list from the GitHub API for a search?

**Comment by lee-dohm** [comment]
<aside class="quote no-group" data-username="0xInfection" data-post="5" data-topic="137762">
<div class="title">
<div class="quote-controls"></div>
 0xInfection:</div>
<blockquote>
<p>Is there a way I can grab that list from the GitHub API for a search?</p>
</blockquote>
</aside>
<p>No, that information isn’t exposed directly through the GitHub API. You would have to search and then iterate through the list of individual results to collect that information.</p>
